wdlc.sh.in

```diff
@@ -271,7 +271,7 @@ function remove_proto_descriptor_set() {
 #
 function protoc_generate_code() {
     local template_dir="${1}"
-    local template_files=$(find ${template_dir}/* ! -name "*~" | tr '\n' ':' | sed 's/:$//')
+    local template_files=$(find ${template_dir} ! -type d ! -name "*~" | tr '\n' ':' | sed 's/:$//')
```

Comment:
I would suggest instead of `! -type d` use `-type f` or `-type f,l` (the latter syntax requires a GNU extensions of find)

Reply:
Additional suggestions: 

* replace the `|tr '\n' ':'` with `-printf "%p:"`
* Purely personal preference: I find it easier to not think about the negation of the term; if you choose, you could replace `! -find "*~"` with `-find "*[^~]"`

Reply:
> I would suggest instead of `! -type d` use `-type f` or `-type f,l` (the latter syntax requires a GNU extensions of find)

Links foul up '-type f' which is why I aligned on '! -type d'. I'd rather avoid using a GNU-specific extension.